habitat-lab/habitat/config/CONFIG_KEYS.md

```diff
@@ -0,0 +1,180 @@
+|habitat.task.reward_measure | The name of the Measurement that will correspond to the reward of the robot. This value must be a key present in the dictionary of Measurements in the habitat configuration. |
+|habitat.task.success_measure | The name of the Measurement that will correspond to the success criteria of the robot. This value must be a key present in the dictionary of Measurements in the habitat configuration. If the measurement has a non-zero value, the episode is considered a success. |
+|habitat.task.end_on_success | If True, the episode will end when the success measure indicates success. Otherwise the episode will go on (this is useful when doing hierarchical learning and the robot has to explicitly decide when to change policies)|
+|habitat.task.task_spec |  When doing the `RearrangeCompositeTask-v0` only, will look for a pddl plan of that name to determine the sequence of tasks that need to be completed. The format of the pddl plans files is undocumented.|
+|habitat.task.task_spec_base_path |  When doing the `RearrangeCompositeTask-v0` only, the relative path where the task_spec file will be searched.|
+|habitat.task.spawn_max_dists_to_obj| For `RearrangePickTask-v0` task only. Controls the maximum distance the robot can be spawned from the target object. |
+| habitat.task.base_angle_noise| For Rearrangement tasks only. Controls the standard deviation of the random normal noise applied to the base's rotation angle at the start of an episode.|
+| habitat.task.base_angle| For Rearrangement tasks only. Controls the standard deviation of the random normal noise applied to the base's position at the start of an episode.|
+
+## Visual Agents
+You can inherit a default agent in your configuration.
+The some sample agents are defined [here](./habitat/simulator/agents/) and are reused in many of our benchmarks. To use one of these agents, add
+```
+  - /habitat/simulator/agents@habitat.simulator.agents.main_agent: <the key of the agent>
+```
+to your defaults list. All agents have a camera resolution of 256 by 256.
+| Key | Description |
+| --- | --- |
+|depth_head_agent| An Agent with a single depth camera attached to its head.|
+|rgb_head_agent| An Agent with a single RGB camera attached to its head.|
+|rgbd_head_rgbd_arm_agent| An Agent with both a RGB and depth camera attached to its head and arm.|
+|depth_head_agent_vis| An Agent with a depth camera attached to its head and a third person RGB camera on top of the robot.|
+
+
+
+
+## Environment
+Some habitat environment configurations.
+| Key | Description |
+| --- | --- |
+|habitat.environment.max_episode_steps| The maximum number of environment steps before the episode ends.|
+|habitat.environment.max_episode_seconds| The maximum number of seconds steps before the episode ends.|
+
+## Navigation Actions
+The way one would add an action to a configuration file would be by adding to the `defaults` list. For example:
```

Comment:
can you describe what actions are? Typically, in RL actions are defined by the env, for example if actions are joint angles, torques, end-effector position.
Is this what we are trying to define here? So is this actions dictionary meant to be added to a task config?